scripts/zones/Misareaux_Coast/npcs/qm2.lua

```diff
@@ -0,0 +1,63 @@
+-----------------------------------
+-- Area: Misareaux_Coast
+-- NPC:  ??? (Spawn Ziphius)
+-- !pos 76 -16 534 25
+-----------------------------------
+
+require("scripts/globals/npc_util")
+local ID = require("scripts/zones/Misareaux_Coast/IDs")
+local ZIPHIUS_QM_BASE = 16879919
+local ZIPHIUS         = 16879900
+function onTrade(player,npc,trade)
+    local baited = GetServerVariable("[Ziphius]Bait Trap")
+    
+    -- Trade Slice of Carp
+    if (npcUtil.tradeHas(trade, 16994) and bit.band(baited, bit.lshift(1, (npc:getID() - ZIPHIUS_QM_BASE))) == 0) then
+        baited = bit.bor(baited, bit.lshift(1, (npc:getID() - ZIPHIUS_QM_BASE)))
+        player:tradeComplete()
```

Comment:
If using npcUtil.tradeHas(), which flags matched items to be removed, then use player:confirmTrade() rather than player:tradeComplete().  confirmTrade consumes only the marked items, and returns excess items in the trade window to the player.  tradeComplete removes from player all items put into the trade window.